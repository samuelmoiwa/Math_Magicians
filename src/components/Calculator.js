/* eslint-disable react/self-closing-comp */
/* eslint-disable react/no-unknown-property */
import '../style.css';
import React, { PureComponent } from 'react';

class Calculator extends PureComponent {
  render() {
    return (
      <div className="calculator_div">
        <div class="calculator_main_container">

          <div class="answer">
            <h2 class="_answer">0</h2>
          </div>

          <div class="colum_1">
            <h3 class="ac">AC</h3>
            <h3 class="add_sub">+/-</h3>
            <h3 class="percentage">%</h3>
            <h3 class="division">รท</h3>
          </div>

          <div class="colum_2">
            <h3 class="_7">7</h3>
            <h3 class="_8">8</h3>
            <h3 class="_9">9</h3>
            <h3 class="_x">X</h3>
          </div>

          <div class="colum_3">
            <h3 class="_4">4</h3>
            <h3 class="_5">5</h3>
            <h3 class="_6">6</h3>
            <h3 class="subtraction">-</h3>
          </div>

          <div class="colum_4">
            <h3 class="_1">1</h3>
            <h3 class="_2">2</h3>
            <h3 class="_3">3</h3>
            <h3 class="addition">+</h3>
          </div>

          <div class="colum_5">
            <h3 class="_0">0</h3>
            <h3 class="dot">.</h3>
            <h3 class="equal">=</h3>
          </div>

        </div>
      </div>
    );
  }
}

export default Calculator;
