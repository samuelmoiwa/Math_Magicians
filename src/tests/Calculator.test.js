import { render } from '@testing-library/react';
import Calculator from '../components/pages/Calculator';

describe('test for creating the calculator', () => {
  it('created calculator', () => {
    const view = render(<Calculator />);
    expect(view).toMatchSnapshot();
  });
});
