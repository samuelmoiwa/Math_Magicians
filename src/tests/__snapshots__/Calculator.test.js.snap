// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for creating the calculator created calculator 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calculator_div"
      >
        <div
          class="lets_do_maths_div"
        >
          <h2
            class="lets_do_maths_text"
          >
             Lets do some math!
          </h2>
          <img
            alt="maths"
            class="image_file"
            src="./maths_image.png"
          />
        </div>
        <div
          class="calculator_main_container"
        >
          <div
            class="answer"
          >
            <input
              class="_answer"
              readonly=""
              type="text"
              value="0"
            />
          </div>
          <div
            class="colum_1"
          >
            <h3
              class="ac"
            >
              AC
            </h3>
            <h3
              class="add_sub"
            >
              +/-
            </h3>
            <h3
              class="percentage"
            >
              %
            </h3>
            <h3
              class="division"
            >
              รท
            </h3>
          </div>
          <div
            class="colum_2"
          >
            <h3
              class="_7"
            >
              7
            </h3>
            <h3
              class="_8"
            >
              8
            </h3>
            <h3
              class="_9"
            >
              9
            </h3>
            <h3
              class="_x"
            >
              x
            </h3>
          </div>
          <div
            class="colum_3"
          >
            <h3
              class="_4"
            >
              4
            </h3>
            <h3
              class="_5"
            >
              5
            </h3>
            <h3
              class="_6"
            >
              6
            </h3>
            <h3
              class="subtraction"
            >
              -
            </h3>
          </div>
          <div
            class="colum_4"
          >
            <h3
              class="_1"
            >
              1
            </h3>
            <h3
              class="_2"
            >
              2
            </h3>
            <h3
              class="_3"
            >
              3
            </h3>
            <h3
              class="addition"
            >
              +
            </h3>
          </div>
          <div
            class="colum_5"
          >
            <h3
              class="_0"
            >
              0
            </h3>
            <h3
              class="dot"
            >
              .
            </h3>
            <h3
              class="equal"
            >
              =
            </h3>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calculator_div"
    >
      <div
        class="lets_do_maths_div"
      >
        <h2
          class="lets_do_maths_text"
        >
           Lets do some math!
        </h2>
        <img
          alt="maths"
          class="image_file"
          src="./maths_image.png"
        />
      </div>
      <div
        class="calculator_main_container"
      >
        <div
          class="answer"
        >
          <input
            class="_answer"
            readonly=""
            type="text"
            value="0"
          />
        </div>
        <div
          class="colum_1"
        >
          <h3
            class="ac"
          >
            AC
          </h3>
          <h3
            class="add_sub"
          >
            +/-
          </h3>
          <h3
            class="percentage"
          >
            %
          </h3>
          <h3
            class="division"
          >
            รท
          </h3>
        </div>
        <div
          class="colum_2"
        >
          <h3
            class="_7"
          >
            7
          </h3>
          <h3
            class="_8"
          >
            8
          </h3>
          <h3
            class="_9"
          >
            9
          </h3>
          <h3
            class="_x"
          >
            x
          </h3>
        </div>
        <div
          class="colum_3"
        >
          <h3
            class="_4"
          >
            4
          </h3>
          <h3
            class="_5"
          >
            5
          </h3>
          <h3
            class="_6"
          >
            6
          </h3>
          <h3
            class="subtraction"
          >
            -
          </h3>
        </div>
        <div
          class="colum_4"
        >
          <h3
            class="_1"
          >
            1
          </h3>
          <h3
            class="_2"
          >
            2
          </h3>
          <h3
            class="_3"
          >
            3
          </h3>
          <h3
            class="addition"
          >
            +
          </h3>
        </div>
        <div
          class="colum_5"
        >
          <h3
            class="_0"
          >
            0
          </h3>
          <h3
            class="dot"
          >
            .
          </h3>
          <h3
            class="equal"
          >
            =
          </h3>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
